diff --git a/gcc-4.6.1/gcc/config/host-linux.c b/gcc-4.6.1/gcc/config/host-linux.c
index 47ce3ea..98b5586 100644
--- a/gcc-4.6.1/gcc/config/host-linux.c
+++ b/gcc-4.6.1/gcc/config/host-linux.c
@@ -23,6 +23,10 @@
 #include "hosthooks.h"
 #include "hosthooks-def.h"
 
+#ifndef SSIZE_MAX
+# define SSIZE_MAX (SIZE_MAX / 2)
+#endif /* SSIZE_MAX */
+
 
 /* Linux has a feature called exec-shield-randomize that perturbs the
    address of non-fixed mapped segments by a (relatively) small amount.
